Sources/NIOPosix/Bootstrap.swift

```diff
@@ -482,6 +482,484 @@ public final class ServerBootstrap {
+        }
+
+        return try await self.bindAsyncChannelWithProtocolNegotiation0(
+            makeServerChannel: makeChannel,
+            protocolNegotiationHandlerType: protocolNegotiationHandlerType,
+            serverBackpressureStrategy: serverBackpressureStrategy
+        ) { (eventLoop, serverChannel) in
+            serverChannel.registerAndDoSynchronously { serverChannel in
+                serverChannel.bind(to: address)
+            }
+        }
+    }
+
+    @available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
+    private func bindAsyncChannelWithProtocolNegotiation0<Handler: NIOProtocolNegotiationHandler & ChannelHandler>(
+        makeServerChannel: MakeServerChannel,
+        protocolNegotiationHandlerType: Handler.Type,
+        serverBackpressureStrategy: NIOAsyncSequenceProducerBackPressureStrategies.HighLowWatermark?,
+        _ register: @escaping Register
+    ) async throws -> NIOAsyncChannel<Handler.NegotiationResult, Never>  {
+        let eventLoop = self.group.next()
+        let childEventLoopGroup = self.childGroup
+        let serverChannelOptions = self._serverChannelOptions
+        let serverChannelInit = self.serverChannelInit ?? { _ in eventLoop.makeSucceededFuture(()) }
+        let childChannelInit = self.childChannelInit
+        let childChannelOptions = self._childChannelOptions
+
+        let serverChannel = try makeServerChannel(eventLoop as! SelectableEventLoop, childEventLoopGroup, self.enableMPTCP)
+
+        return try await eventLoop.submit {
+            serverChannelOptions.applyAllChannelOptions(to: serverChannel).flatMap {
+                serverChannelInit(serverChannel)
+            }.flatMap {
+                do {
+                    try serverChannel.pipeline.syncOperations.addHandler(
+                        AcceptHandler(childChannelInitializer: childChannelInit, childChannelOptions: childChannelOptions),
+                        name: "AcceptHandler"
+                    )
+
+                    // In the case of protocol negotiation we cannot wrap the child channels into
+                    // `NIOAsyncChannel`s for the user since we don't know the type. We rather exepect
+                    // the user to wrap the child channels themselves when the negotiation is done
+                    // and return the wrapped async channel as part of the negotiation result.
+                    let asyncChannel = try NIOAsyncChannel<Handler.NegotiationResult, Never>(
+                        synchronouslyWrapping: serverChannel,
+                        backpressureStrategy: serverBackpressureStrategy,
+                        protocolNegotiationClosure: { (channel: Channel) in
+                            channel.eventLoop
+                                .makeCompletedFuture {
+                                    try channel.pipeline.syncOperations.handler(type: protocolNegotiationHandlerType)
+                                }
```

Comment:
Isn't this just `channel.pipelline.handler(type: protocolNegotiationHandlerType)`?